.github/workflows/ci.yml

```diff
@@ -74,6 +74,7 @@ jobs:
         pip install -U pip
         # Install requirements, using eager updates to avoid stalled dependencies due to caching
         # https://blog.allenai.org/python-caching-in-github-actions-e9452698e98d
+        pip install --upgrade --upgrade-strategy eager -r requirements.txt
```

Comment:
We might want to split the `Install Porepy` stage into three parts: (1) one where the porepy dependencies from `requirements.txt` are installed, (2) another where the porepy dependencies from `requirements_dev.txt` are installed, and (3) another where porepy is actually installed.

This will allow us to isolate build failures which are strictly related to those steps, i.e., install of the requirements, install of the dev requirements, and install of porepy itself.

What are your thoughts about this suggestion @OmarDuran @keileg?

Reply:
@jhabriel: How is that different from lines 77-79 now?

Reply:
The  only difference is that Actions will treat the 3 processes separately. So, if there is an issue installing some package from `requirements.txt`, it will show that this action failed (rather than the whole porepy installation), which IMO will allow us to isolate potential dependencies issues more easily.

In practice is no different from what we have now in lines 77-79.

Reply:
Ah, I see the point. Fair enough, let's split it.

Reply:
The installation has been split PorePy installation into three actions.